### PR TITLE
build: make boa work with Node10 and NAPI3

### PR DESCRIPTION
**Pull request by yorkie**
Node.js 12 requires a higher GLIBC, which won't be supported on some platforms, thus this PR downgrades the Node.js and NAPI requirements. Now the boa is able to work with Node.js 10.13 and NAPI3.

### CODE REVIEW

**Comment by yorkie** [comment]
@FeelyChau The CI is green :)

File: packages/boa/package.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
     "eslint": "^7.15.0",
     "nyc": "^15.1.0",
     "tape": "^5.0.1",
-    "ts-node": "^8.6.2"
+    "ts-node": "^8.6.2",
+    "typescript": "^4.1.3"
```
Comment (FeelyChau) [inline packages/boa/package.json:57]: Why import the typescript?

File: packages/boa/lib/worker.js [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
+let isMainThread = false;
+let workerData = null;
+let supportWorkerThreads = false;
+
 const { PythonObject, NODE_PYTHON_HANDLE_NAME } = require('bindings')('boa');
 const { wrap } = require('./proxy');
 
 const TYPE_ID = 'SHARED_PYTHON_OBJECT_TYPE';
+try {
+  require.resolve('worker_threads');
```
Comment (FeelyChau) [inline packages/boa/lib/worker.js:12]: Could we do the flag though the version of process instead trying?

**Review by FeelyChau — CHANGES_REQUESTED** [review]

File: packages/boa/lib/worker.js (hunk 2 above)
Comment (yorkie) [reply packages/boa/lib/worker.js:12]: I think a feature-based check is better than a version-based.

File: packages/boa/package.json (hunk 1 above)
Comment (yorkie) [reply packages/boa/package.json:57]: Oops.

**Review by FeelyChau — APPROVED** [review]
LGTM.

**Review by rickycao-qy — APPROVED** [review]
LGTM

**Pull request merged**